### PR TITLE
Drop remaining primitive FieldValue types

### PR DESCRIPTION
**Pull request by schmidt-sebastian**

### CODE REVIEW

File: firebase-firestore/src/main/java/com/google/firebase/firestore/model/value/ProtoValues.java [hunk 1]
```diff
@@ -41,34 +41,47 @@
   public static final Value NULL_VALUE =
       Value.newBuilder().setNullValue(NullValue.NULL_VALUE).build();
 
+  /** The order of types in Firestore; this order is defined by the backend. */
+  private static final int TYPE_ORDER_NULL = 0;
```
Comment (wilhuff) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/model/value/ProtoValues.java]: public methods that return private values make little sense. `typeOrder` should either be private or these constants should be public.

File: firebase-firestore/src/test/java/com/google/firebase/firestore/UserDataWriterTest.java [hunk 2]
```diff
@@ -291,4 +287,8 @@ public void testRejectsJavaArrays() {
   private Object convertValue(FieldValue value) {
     return writer.convertValue(value.getProto());
   }
+
+  private void assertValueType(FieldValue value, Value.ValueTypeCase booleanValue) {
```
Comment (wilhuff) [inline firebase-firestore/src/test/java/com/google/firebase/firestore/UserDataWriterTest.java]: JUnit tends to put the expected value as the first argument. Consider reversing these to match.

File: firebase-firestore/src/test/java/com/google/firebase/firestore/UserDataWriterTest.java [hunk 3]
```diff
@@ -259,12 +257,10 @@ private static ObjectValue fromMap(Object... entries) {
...
   public void testConvertsNestedObjects() {
     FieldValue actual = wrapObject("a", map("b", map("c", "foo"), "d", true));
-    ObjectValue expected =
-        fromMap(
-            "a",
-            fromMap(
-                "b", fromMap("c", StringValue.valueOf("foo")), "d", BooleanValue.valueOf(true)));
-    assertEquals(expected, actual);
+    ObjectValue.Builder expected = ObjectValue.newBuilder();
+    expected.set(field("a.b.c"), valueOf("foo"));
```
Comment (wilhuff) [inline firebase-firestore/src/test/java/com/google/firebase/firestore/UserDataWriterTest.java]: Optional: You could chain these (`ObjectValue expected = newBuilder.set.set.build()`)

File: firebase-firestore/src/main/java/com/google/firebase/firestore/model/value/ProtoValues.java (hunk 1 above)
Comment (schmidt-sebastian) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/model/value/ProtoValues.java]: Made them public.

File: firebase-firestore/src/test/java/com/google/firebase/firestore/UserDataWriterTest.java (hunk 2 above)
Comment (schmidt-sebastian) [reply firebase-firestore/src/test/java/com/google/firebase/firestore/UserDataWriterTest.java]: Changed this. Note that I have a similar helper for canonical IDs, which I kept the same as the callsites look much cleaner in reverse argument order.

File: firebase-firestore/src/test/java/com/google/firebase/firestore/UserDataWriterTest.java (hunk 3 above)
Comment (schmidt-sebastian) [reply firebase-firestore/src/test/java/com/google/firebase/firestore/UserDataWriterTest.java]: It felt cleaner to spread these assignments over multiple lines. Kept as is.

**Pull request merged**